### PR TITLE
new static vertex READ

### PR DESCRIPTION
**Pull request by mr-phazer**
I have changed the Read function for static vertices, now it  reads the normals, tangents, bitangent correctly, but it is does not roate the model.

try it out, it seems to work well.

I have kept the other read method called it ___old_read, because I needed a reference because of my poor C# knowledge:)

### CODE REVIEW

File: CommonControls/FileTypes/RigidModel/Vertex/Formats/StaticVertex.cs [hunk 1]
```diff
@@ -20,7 +20,8 @@ public uint GetVertexSize(RmvVersionEnum rmvVersion)
         }
         public bool ForceComputeNormals => false;
 
-        public CommonVertex Read(RmvVersionEnum rmvVersion, byte[] buffer, int offset, int vertexSize)
+        // preserved for me to look at, for reference :)
+        public CommonVertex ___old_Read(RmvVersionEnum rmvVersion, byte[] buffer, int offset, int vertexSize)
```
Comment (olekristianhomelien) [inline CommonControls/FileTypes/RigidModel/Vertex/Formats/StaticVertex.cs:24]: delete unreferenced code

File: CommonControls/FileTypes/RigidModel/Vertex/Formats/StaticVertex.cs [hunk 2]
```diff
@@ -42,6 +43,39 @@ public CommonVertex Read(RmvVersionEnum rmvVersion, byte[] buffer, int offset, i
             return vertex;
         }
 
+        static Vector3 _swapXZ(Vector3 v)
```
Comment (olekristianhomelien) [inline CommonControls/FileTypes/RigidModel/Vertex/Formats/StaticVertex.cs:46]: Should be SwapXY, underscore not used for functions.
Also consider static Vector3 SwapXZ(Vector3 v) => new vector3(...)

File: CommonControls/FileTypes/RigidModel/Vertex/Formats/StaticVertex.cs [hunk 3]
```diff
@@ -42,6 +43,39 @@ public CommonVertex Read(RmvVersionEnum rmvVersion, byte[] buffer, int offset, i
...
+            var vertex = new CommonVertex()
+            {
+             
+                Position = VertexLoadHelper.CreatVector4HalfFloat(vertexData.position).ToVector4(1),
+
+                /*  VertexFormat = ´default` format has X and Z swapped 
+                    'bitangent' is stored before 'tangent' when VertexFormat = ´default` 
+                */
+                Normal = _swapXZ(VertexLoadHelper.CreatVector4Byte(vertexData.normal).ToVector3()),
+                BiNormal = _swapXZ(VertexLoadHelper.CreatVector4Byte(vertexData.tangent).ToVector3()),
```
Comment (olekristianhomelien) [inline CommonControls/FileTypes/RigidModel/Vertex/Formats/StaticVertex.cs:63]: One can swap the tanget and binormal in the vertexData object to make it less confusing

File: CommonControls/FileTypes/RigidModel/Vertex/Formats/StaticVertex.cs [hunk 4]
```diff
@@ -42,6 +43,39 @@ public CommonVertex Read(RmvVersionEnum rmvVersion, byte[] buffer, int offset, i
...
+                BoneIndex = new byte[0],
+                BoneWeight = new float[0]
+            };
+
+            return vertex;
+        }
+
+
+
         public byte[] Write(RmvVersionEnum rmvVersion, CommonVertex vertex)
```
Comment (olekristianhomelien) [inline CommonControls/FileTypes/RigidModel/Vertex/Formats/StaticVertex.cs:79]: This function also needs to be updated

**Review by olekristianhomelien — CHANGES_REQUESTED** [review]

File: CommonControls/FileTypes/RigidModel/Vertex/Formats/StaticVertex.cs (hunk 3 above)
Comment (mr-phazer) [reply CommonControls/FileTypes/RigidModel/Vertex/Formats/StaticVertex.cs:63]: You want code below this object initilization, that does the swap instead? Or?

**Review by olekristianhomelien — APPROVED** [review]

**Pull request merged**